shotover-proxy/src/server.rs

```diff
@@ -436,6 +445,8 @@ impl<C: Codec + 'static> Handler<C> {
         while !self.shutdown.is_shutdown() {
             // While reading a request frame, also listen for the shutdown signal
             debug!("Waiting for message");
+            let mut reverse_chain = false;
```

Comment:
We should be able to rewrite without this extra boolean state:
![image](https://user-images.githubusercontent.com/5120858/172769889-072bd242-035e-41db-af02-b0ff8511c0d4.png)

Reply:
oh wait, my suggestion moves the chain processing within the timeout!
I dont think thats a good idea.

Reply:
Should this be resolved then?